core/vm/evm.go

```diff
@@ -452,11 +452,13 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 	if evm.StateDB.GetNonce(address) != 0 || (contractHash != (common.Hash{}) && contractHash != emptyCodeHash) {
 		return nil, common.Address{}, 0, ErrContractAddressCollision
 	}
-	// Check that [tx.origin] has the right to deploy contracts
-	modifyStatus := getAllowListStatus(evm.StateDB, evm.Origin)
-	if !modifyStatus.HasDeployPrivileges() {
-		log.Info("origin is not authorized to deploy a contract", "origin", evm.Origin, "status", modifyStatus)
-		return nil, common.Address{}, 0, fmt.Errorf("origin %s is not authorized to deploy a contract", evm.Origin)
+	if getAllowListEnabled(evm.StateDB) {
```

Comment:
I think we should use `chainConfig` here instead to decide whether or not this should be treated as enabled to follow the existing pattern. This should be easier to maintain for optional stateful precompiles going forward.